### DISCUSSION TITLE
I want to know deep link url

### DISCUSSION THREAD

**Discussion by matthew2099** [WalletConnect/WalletConnectUnity#189] [Q&A]
Just i want to metamask and oks url

Jst i write this code

`using System.Collections.Generic;
using UnityEngine;
using WalletConnectSharp.Sign.Models;
using WalletConnectSharp.Sign.Models.Engine;
using WalletConnectUnity.Core;

public class Test : MonoBehaviour
{
    async void Start()
    {
        Init();
    }
    
    public async void Init()
    {
        await WalletConnect.Instance.InitializeAsync();
        
        var optionalNamespaces = new Dictionary<string, ProposedNamespace>();
        
        var methods = new[]
        {
            "wallet_switchEthereumChain",
            "wallet_addEthereumChain",
            "eth_sendTransaction",
            "personal_sign"
        };

        var events = new[]
        {
            "chainChanged", "accountsChanged"
        };

        string[] chainID = { "1001" };

        optionalNamespaces.Add(ChainConstants.Namespaces.Evm, new ProposedNamespace
        {
            Chains = chainID,
            Events = events,
            Methods = methods
        });
        
        var connectOptions = new ConnectOptions
        {
            OptionalNamespaces = optionalNamespaces
        };
        
        var sessionResumed = await WalletConnect.Instance.TryResumeSessionAsync();              
        if (!sessionResumed)                                                                         
        {                                                                                            
            var connectedData = await WalletConnect.Instance.ConnectAsync(connectOptions);
            // ... Display QR code texture
        
            // Wait for wallet approval
            await connectedData.Approval;                                                            
        }                            
    }
}`

I don't know how to specify the wallet.
hum...what should i do?

I want to use deep link.

**Comment by matthew2099** [comment]
`await WalletConnect.Instance.InitializeAsync();

        var optionalNamespaces = new Dictionary<string, ProposedNamespace>();

        var methods = new[]
        {
            "wallet_switchEthereumChain",
            "wallet_addEthereumChain",
            "eth_sendTransaction",
            "personal_sign"
        };

        var events = new[]
        {
            "chainChanged", "accountsChanged"
        };

        string[] chainID = { "1001" };

        optionalNamespaces.Add(ChainConstants.Namespaces.Evm, new ProposedNamespace
        {
            Chains = chainID,
            Events = events,
            Methods = methods
        });

        var options = new WalletConnectModalOptions
        {
            ConnectOptions = new ConnectOptions
            {
                OptionalNamespaces = optionalNamespaces
            },
            ExcludedWalletIds = new[]
            {
                //"c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96", // MetaMask
                "971e689d0a5be527bac79629b4ee9b925e82208e5168b733496a09c0faed0709", // okx
            }
        };`

Now my code.
What should i do next?

just i want deep link

**Comment by skibitsky** [comment]
@matthew2099, do you use WalletConnect Modal or WalletConnect Core package?

**Reply by matthew2099** [reply]
Ah. I use both. and my example source is related about Modal.
Maybe. if i use Core package, is this possible?

**Reply by skibitsky** [reply]
The Modal handles deep linking automatically.

You can also use Core package to open deep link on session proposal and session request, but you will need to manually get the link from the explorer and make sure it's used when necessary.

I recommend using the Modal package. It's basically the Core package with cross-platform UI and some additional features such as deep linking.

**Reply by matthew2099** [reply]
@skibitsky 
Hum...ah i fixed my source.
And i think i make a deep linking. But my deep link don't active..
What's wrong??

`await WalletConnect.Instance.InitializeAsync();

        var optionalNamespaces = new Dictionary<string, ProposedNamespace>();

        var methods = new[]
        {
            "wallet_switchEthereumChain",
            "wallet_addEthereumChain",
            "eth_sendTransaction",
            "personal_sign"
        };

        var events = new[]
        {
            "chainChanged", "accountsChanged"
        };

        string[] chainID = { "1001" };

        optionalNamespaces.Add(ChainConstants.Namespaces.Evm, new ProposedNamespace
        {
            Chains = chainID,
            Events = events,
            Methods = methods
        });

        var connectOptions = new ConnectOptions
        {
            OptionalNamespaces = optionalNamespaces
        };

        // var options = new WalletConnectModalOptions
        // {
        //     ConnectOptions = connectOptions,
        //     ExcludedWalletIds = new[]
        //     {
        //         //"c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96", // MetaMask
        //         "971e689d0a5be527bac79629b4ee9b925e82208e5168b733496a09c0faed0709", // okx
        //     }
        // };

        var connectedData = await WalletConnect.Instance.ConnectAsync(connectOptions);

        Linker.OpenSessionProposalDeepLink(connectedData.Uri, new Wallet()
        {
            Id = "971e689d0a5be527bac79629b4ee9b925e82208e5168b733496a09c0faed0709",
            Name = "OKX Wallet",
            Homepage = "https://www.okx.com/web3",
            ImageId = "45f2f08e-fc0c-4d62-3e63-404e72170500",
            WebappLink = "https://www.okx.com/download",
            AppStore = " https://apps.apple.com/us/app/okx-buy-bitcoin-eth-crypto/id1327268470",
            PlayStore = "https://play.google.com/store/apps/details?id=com.okinc.okex.gp"
        });`

**Reply by skibitsky** [reply]
Do you get any error after calling `Linker.OpenSessionProposalDeepLink`?

**Reply by matthew2099** [reply]
@skibitsky 
I don't see error..
So i don't know what's wrong...

**Reply by skibitsky** [reply]
Do you test it on Android device?

**Reply by matthew2099** [reply]
@skibitsky 
Ah i see the error.
Exception: [Linker] No link found for Android platform in wallet OKX Wallet.
But i installed the okx wallet.
And i see The modal exmaple code is active.

**Reply by matthew2099** [reply]
@skibitsky 
I know that...
I missing the mobile url...

**Reply by matthew2099** [reply]
@skibitsky 
I success that.
But I see the error that "Network error and try again"...haha
Maybe I think chain setting error.
Thank you.

**Reply by skibitsky** [reply]
That may be OKX Wallet issue, sometimes it works if you open deep link again.

By the way, why do you want to only support OKX Wallet?

**Reply by matthew2099** [reply]
@skibitsky 
But i have a question.
Rarely I see the Connection Success. But usually i don't see that...is normal??

**Reply by matthew2099** [reply]
@skibitsky 
Ah my company's coin is related to okx wallet.
So i must use okx wallet....

**Reply by skibitsky** [reply]
> Rarely I see the Connection Success. But usually i don't see that...is normal??

It isn't normal. I'm investigating that and will be posting updates under this issue. https://github.com/WalletConnect/WalletConnectUnity/issues/185
